### PR TITLE
Add new test for None values on string columns on a Dataframe

### PR DESCRIPTION
**Pull request by GrammatikakisDimitris**
- [x]  #32218
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Although this issue seems to be gone, i noticed in the comments that a test for this would be appreciated, so this basically checks that Dataframe behaves as it should.

### CODE REVIEW

File: pandas/tests/frame/test_constructors.py [hunk 1]
```diff
@@ -2698,6 +2698,14 @@ def test_frame_from_dict_with_mixed_tzaware_indexes(self):
 
 
 class TestDataFrameConstructorWithDtypeCoercion:
+    def test_none_values_on_string_columns(self):
```
Comment (mroeschke) [inline pandas/tests/frame/test_constructors.py:2701]: Can you move this to `pandas/tests/indexing/test_loc.py`?

File: pandas/tests/frame/test_constructors.py [hunk 2]
```diff
@@ -2698,6 +2698,14 @@ def test_frame_from_dict_with_mixed_tzaware_indexes(self):
...
+        # Creating DataFrame with column "a" as string and a None value
```
Comment (mroeschke) [inline pandas/tests/frame/test_constructors.py:2702]: These comments are not needed. Instead could you add a comment with the github issue number?

File: pandas/tests/frame/test_constructors.py [hunk 3]
```diff
@@ -2698,6 +2698,14 @@ def test_frame_from_dict_with_mixed_tzaware_indexes(self):
...
+        data = {"a": ["1", "2", None]}
+        df = DataFrame(data=data, dtype="str")
+
+        # Checking that the value of the third row in column "a" is actually None
+        assert isna(df.loc[2, "a"])
```
Comment (mroeschke) [inline pandas/tests/frame/test_constructors.py:2707]:
```suggestion
        assert df.loc[2, "a"] is None
```

Comment (GrammatikakisDimitris) [reply pandas/tests/frame/test_constructors.py]: Greetings @mroeschke, i made the changes as requested

File: pandas/tests/indexing/test_loc.py [hunk 4]
```diff
@@ -62,6 +62,13 @@ def test_not_change_nan_loc(series, new_series, expected_ser):
 
 
 class TestLoc:
+    def test_none_values_on_string_columns(self):
+        # Issue #32218
+        data = {"a": ["1", "2", None]}
+        df = DataFrame(data=data, dtype="str")
```
Comment (mroeschke) [inline pandas/tests/indexing/test_loc.py:68]: From the issue `df = pd.DataFrame(["1", "2", None], columns=["a"], dtype="str")` should be the DataFrame being tested here

Comment (GrammatikakisDimitris) [reply pandas/tests/indexing/test_loc.py:68]: Pre-commit check fails if i write pd.Dataframe, is it ok to write Dataframe instead?

Comment (mroeschke) [reply pandas/tests/indexing/test_loc.py:68]: Yes

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @GrammatikakisDimitris

**Pull request merged**